### PR TITLE
637 varinfo lfric builtins

### PR DESCRIPTION
**Pull request by hiker**
A somewhat hacky solution, not using any PSyIR,.

### CODE REVIEW

**Comment by hiker** [comment]
Ready for first review.

File: examples/lfric/eg17/full_example_extract/extract_transform.py [hunk 1]
```diff
@@ -62,13 +62,12 @@ def trans(psy):
...
     extract = ExtractTrans()
 
-    # We don't support builtins yet, so the initialisation
-    # cannot be instrumented, TODO #637
-    # invoke = psy.invokes.get("invoke_initialise_fields")
-    # schedule = invoke.schedule
-    # extract.apply(schedule.children,
-    #               {"create_driver": True,
-    #                "region_name": ("main", "init")})
+    # Show that it works on a builtin:
```
Comment (arporter) [inline examples/lfric/eg17/full_example_extract/extract_transform.py:65]: Whoop :-)

File: src/psyclone/domain/lfric/lfric_builtins.py [hunk 2]
```diff
@@ -166,6 +166,33 @@ def __init__(self):
...
+            if arg.form == "variable":
+                if arg.access == AccessType.WRITE:
+                    written.add_access(Signature(arg.declaration_name),
+                                       arg.access, self)
+                else:
+                    var_accesses.add_access(Signature(arg.declaration_name),
+                                            arg.access, self)
+        # Now merge the write access to the end of all other accesses:
+        var_accesses.merge(written)
+        # Forwart location pointer to next index, since this builtin kernel
```
Comment (arporter) [inline src/psyclone/domain/lfric/lfric_builtins.py:192]: "Forward"

File: src/psyclone/tests/domain/lfric/transformations/lfric_extract_test.py [hunk 3]
```diff
@@ -549,23 +563,12 @@ def test_extract_single_builtin_dynamo0p3():
       !$omp end parallel do
       CALL extract_psy_data%PostStart
       CALL extract_psy_data%ProvideVariable("df_post", df)
+      CALL extract_psy_data%ProvideVariable("f1_post", f1)
       CALL extract_psy_data%PostEnd
       !
       ! ExtractEnd"""
     assert output in code_omp
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/transformations/lfric_extract_test.py:570]: Can we add compilation testing to this test?
Edit: not yet - see #706

**Review by arporter — APPROVED** [review]
All good apart from one tiny typo which I shall fix.
All tests and examples are OK with compilation (and the updated eg17 works when executed).
Ref. guide builds fine, as does updated UG.
Changes are pylint clean.
I'll proceed to merge...

File: src/psyclone/domain/lfric/lfric_builtins.py (hunk 2 above)
Comment (arporter) [reply src/psyclone/domain/lfric/lfric_builtins.py]: Fixed myself.

**Pull request merged**